app/controllers/static_pages_controller.rb

```diff
@@ -1,7 +1,56 @@
-class StaticPagesController < ApplicationController
-  def home
+require 'tsort'
+
+# Topological sorting hash
+class TsortableHash < Hash
+  include TSort
+  alias tsort_each_node each_key
+  def tsort_each_child(node, &block)
+    fetch(node).each(&block)
   end
+end
+
+class StaticPagesController < ApplicationController
+  def formula
+    @projects = Project.all
+    @data_types = DataType.all
+    @data_values = DataValue.all
+
+    # Find dependencies of each data value
+    data_values = TsortableHash[]
+    @data_values.each do |data_value|
+      formula = @data_types.find(data_value.data_type_id).formula
+      if formula
+        year_id = data_value.year_id
+        dependencies = []
+        @data_types.each do |data_type|
+          if formula.include? data_type.name
+            dependencies.push(@data_values.find_by(year_id: year_id, data_type_id: data_type.id))
+          end
+        end
+        data_values[data_value] = dependencies
+      else
+        data_values[data_value] = []
+      end
+    end
```

Comment:
Definitely! In fact, arguably it's even more fit for a service object (which is just a fancy word for a plain Ruby class) if it depends on multiple models (since we try to minimize coupling on the model level and computational logic on the controller level).